.ci/go-lint.sh

```diff
@@ -25,16 +25,15 @@ then
 fi
 
 gometalinter \
-	--exclude='error return value not checked.*(Close|Log|Print).*\(errcheck\)$' \
-	--exclude='.*_test\.go:.*error return value not checked.*\(errcheck\)$' \
-	--exclude='duplicate of.*_test.go.*\(dupl\)$' \
-	--disable=aligncheck \
-	--disable=gotype \
-	--disable=gas \
-	--disable=vetshadow \
-	--cyclo-over=15 \
 	--tests \
-	--deadline=600s \
 	--vendor \
-	--errors \
+	--disable-all \
```

Comment:
This worries me slightly since as `gometalinter` grows more functionality over time, we won't be automatically taking advantage of it due to this flag. However, looking at the project docs it seems this is the preferred way to handle running it under a CI.

What we could do is something like:

```
    :

# When running the linters in a CI environment we need to disable them all by default
# and then explicitly enable the ones we are care about. This is necessary since *iff* gometalinter
# adds a new linter, that linter may cause the CI build to fail when it really shouldn't. However, when
# this script is run locally, all linters should be run to allow the developer to review any failures
# (and potentially decide whether we need to explicitly enable a new linter in the CI).
if [ "$CI" = true ]; then
    linter_args+=" --disable-all"
fi

linter_args+=" --enable=misspell"
linter_args+=" --enable=vet"

    :
```

That would DTRT under CI, but would run all linters I think(?) for devs calling `.ci/go-lint.sh` manually as we should all be doing before raising a PR. Yes, if new linters get added they might error locally, but that is a signal the code needs improving (and that we need to potentially explicitly enable that new linter in `.ci/go-lint.sh`).